bindings/c/test/mako/ddsketch.hpp

```diff
@@ -188,7 +201,7 @@ class DDSketchBase {
 	double errorGuarantee; // As defined in the paper
 
 	uint64_t populationSize, zeroPopulationSize; // we need to separately count 0s
-	std::vector<uint64_t> buckets;
+	std::vector<uint32_t> buckets;
```

Comment:
How much space do we save?

Reply:
We basically halve the amount of memory. I originally intended to run with 32bit unsigned but it appears I left accidentally put it as 64 bit. 

With 32 bit:
`8300 buckets * 4 bytes * 16 operations * 8 processes * 256 threads ~ 1.08GB`

so with 64 bits it would be 2GB. However, perfsanity can run up to 512 threads. In that case, the difference becomes much bigger (2.17 GB vs 4.3 GB)

Reply:
Let's keep uint64_t such that we never hit the issue above.

Reply:
What was the rational behind not sharing DDSketch structures among threads? (or even processes?)

Reply:
Short answer: Confusing behavior with linux shared memory led to it not working properly.

Longer answer: I initially tried to use the linux `shm` code to simply make one DDSketch that can be shared across all processes (and also threads). However due to the way the previous stats information was collected (on a per thread basis) it required me to tweak around with shared memory. 

I had ran into a situation where the buckets data from the worker processes wasn't being properly updated. Specifically, whenever a worker process incremented a bucket, it would still end up being zero on the stats process side. I spent about 2 days debugging and didn't really make much progress. That's when I decided to leave it as is for the sake of time. 

In a perfect world, there would be 16 DDSketches (one for each operations) and the processes would never have any race conditions when adding entries that map to the same bucket. Unfortunately, with the way it was previously set up in addition to the fact that I am not that familiar with IPCs and shared memory I currently have this approach.